sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample03_EventHubMetadata.md

```diff
@@ -0,0 +1,90 @@
+# Event Hub Metadata
+
+This sample discusses the metadata available for an Event Hub instance and demonstrates how to query and inspect the information.  To begin, please ensure that you're familiar with the items discussed in the [Getting started](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventhub/Azure.Messaging.EventHubs/samples#getting-started) section of the README, and have the prerequisites and connection string information available.
+
+# Client types
+
+Querying and inspecting metadata is a common scenario when publishing and reading events.  As a result, the core operations are available to the `EventHubProducerClient` and `EventHubConsumerClient`.  
+
+Both the `EventHubProducerClient` and `EventHubConsumerClient` are safe to cache and use for the lifetime of an application, which is best practice when the application publishes or reads events regularly or semi-regularly. The clients are responsible for efficient resource management, working to keep resource usage low during periods of inactivity and manage health during periods of higher use. Calling either the `CloseAsync` or `DisposeAsync` method on a client as the application is shutting down will ensure that network resources and other unmanaged objects are properly cleaned up.
+
+# Query the properties of an Event Hub
+
+Because the Event Hubs clients operate on a specific Event Hub, it is often helpful for them to have knowledge of its context.  In particular, it is common for clients to understand the partitions available.  The ability to query the Event Hub properties is available using the `EventHubProducerClient` and `EventHubConsumerClient`.  For illustration, the `EventHubProducerClient` is demonstrated, but the concept and form are common across both clients.
+
+```C# Snippet:EventHubs_Sample03_InspectHub
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+var producer = new EventHubProducerClient(connectionString, eventHubName);
+
+try
+{
+    EventHubProperties properties = await producer.GetEventHubPropertiesAsync();
+
+    Debug.WriteLine("The Event Hub has the following properties:");
+    Debug.WriteLine($"\tThe path to the Event Hub from the namespace is: { properties.Name }");
+    Debug.WriteLine($"\tThe Event Hub was created at: { properties.CreatedOn }, in UTC.");
+    Debug.WriteLine($"\tThe following partitions are available: [{ string.Join(", ", properties.PartitionIds) }]");
+}
+finally
+{
+    await producer.CloseAsync();
+}
+```
+
+# Query the partitions of an Event Hub
+
+Due to their importance, there is also a shorthand way to query the partitions of an Event Hub.  The ability to query the Event Hub properties is available using the `EventHubProducerClient` and `EventHubConsumerClient`.  For illustration, the `EventHubProducerClient` is demonstrated, but the concept and form are common across both clients.
```

Comment:
```suggestion
Due to their importance, there is also a shorthand way to query the partitions of an Event Hub.  This capability is available using the `EventHubProducerClient` and `EventHubConsumerClient`.  For illustration, the `EventHubProducerClient` is demonstrated, but the concept and form are common across both clients.
```

Reply:
Applying manually to avoid throwing off my rebase.